### PR TITLE
Convert wdmerger establish_hse to C++

### PR DESCRIPTION
**Pull request by maxpkatz**
## PR checklist

- [x] test suite needs to be run on this PR
- [ ] this PR will change answers in the test suite to more than roundoff level
- [ ] all newly-added functions have docstrings as per the coding conventions
- [ ] the `CHANGES` file has been updated, if appropriate
- [ ] if appropriate, this change is described in the docs

### CODE REVIEW

**Comment by zingale** [comment]
fails to build: http://groot.astro.sunysb.edu/Castro/test-suite/gfortran/2020-11-22-002/index.html

**Comment by maxpkatz** [comment]
Should be fixed

**Comment by zingale** [comment]
tests: http://groot.astro.sunysb.edu/Castro/test-suite/gfortran/2020-11-22-003/index.html

**Comment by maxpkatz** [comment]
The errors on some tests look large, but I've looked at wdmerger_3D closely and as far as I can tell, everything is roundoff at t = 0. And the initial model is only used in problem initialization, so the only explanation I have is that it's a sensitive test to roundoff.

File: Exec/science/wdmerger/initial_model.cpp [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+
+                Real rho_avg = 0.5_rt * (rho[i] + rho[i-1]);
+                p_want = p_last + model.dx * rho_avg * model.g[i];
+
+                eos(eos_input_rt, eos_state);
+
+                drho = (p_want - eos_state.p) / (eos_state.dpdr - 0.5_rt * model.dx * model.g[i]);
+
+                rho[i] = amrex::max(0.9 * rho[i], min(rho[i] + drho, 1.1 * rho[i]));
+                eos_state.rho = rho[i];
```
Comment (zingale) [inline Exec/science/wdmerger/initial_model.cpp:163]: 0.9_rt
1.1_rt

Comment (maxpkatz) [reply Exec/science/wdmerger/initial_model.cpp:163]: Updated

**Review by zingale — APPROVED** [review]

**Pull request merged**